### PR TITLE
fix(web): websocket connection issues

### PR DESCRIPTION
**Pull request by m8vago**
The connection was not closed when we removed the last route, so it kept reconnecting.
The connection setup process on the server side wasn't always ran in the correct order.

### CODE REVIEW

**Comment by m8vago** [comment]
@polaroi8d @robot9706 please check if it's still buggy. I can't reproduce it now.

File: web/crux-ui/src/websockets/websocket-client.ts [hunk 1]
```diff
@@ -72,37 +72,49 @@ class WebSocketClient {
...
   setErrorHandler(handler: WsErrorHandler) {
     this.errorHandler = handler
   }
 
+  private removeRoute(route: WebSocketClientRoute) {
+    const { path } = route
+    this.routes.delete(path)
+
+    if (route.redirectedFrom) {
+      this.redirectedRoutes.delete(route.redirectedFrom)
```
Comment (polaroi8d) [inline web/crux-ui/src/websockets/websocket-client.ts:98]: I think you could also add a debug comment here.

**Review by polaroi8d — APPROVED** [review]
Small additional, nothing extraordinary okay for me.

**Review by robot9706 — COMMENTED** [review]
Running `deploy-dagent.spec.ts:11` with prod builds of both crux and ui the WS problem still occurs for me (not 100% of the time, but still not rare).

The test hangs where it enters the image name and waits for the image list to load, which never does without WS.

Here's the log showing that the sub messages are missing:
```
[Nest] 17352  - 2023. 05. 09. 9:48:37 VERBOSE [JwtAuthGuard] Authorized. Kratos cookie was found legit.
[Nest] 17352  - 2023. 05. 09. 9:48:37   DEBUG [DyoWsAdapter] Connection authorized - ::ffff:127.0.0.1
[Nest] 17352  - 2023. 05. 09. 9:48:37 VERBOSE [DyoWsAdapter] Binding ae276b98-502c-40b2-ae8e-156728c00e24 to /versions/:versionId
[Nest] 17352  - 2023. 05. 09. 9:48:37 VERBOSE [DyoWsAdapter] Binding ae276b98-502c-40b2-ae8e-156728c00e24 to /teams/:teamId/registries
[Nest] 17352  - 2023. 05. 09. 9:48:37 VERBOSE [DyoWsAdapter] Binding ae276b98-502c-40b2-ae8e-156728c00e24 to /deployments/:deploymentId
[Nest] 17352  - 2023. 05. 09. 9:48:37 VERBOSE [DyoWsAdapter] Binding ae276b98-502c-40b2-ae8e-156728c00e24 to /teams/:teamId/nodes
[Nest] 17352  - 2023. 05. 09. 9:48:37 VERBOSE [DyoWsAdapter] Binding ae276b98-502c-40b2-ae8e-156728c00e24 to /nodes/:nodeId
[Nest] 17352  - 2023. 05. 09. 9:48:38 VERBOSE [JwtAuthGuard] Authorized. Kratos cookie was found legit.
[Nest] 17352  - 2023. 05. 09. 9:48:38     LOG [HTTP] 201 POST /api/nodes
[Nest] 17352  - 2023. 05. 09. 9:48:38 VERBOSE [JwtAuthGuard] Authorized. Kratos cookie was found legit.
[Nest] 17352  - 2023. 05. 09. 9:48:38     LOG [HTTP] 201 POST /api/nodes/db678827-69ad-4eff-87a5-ae08575d9040/script
[Nest] 17352  - 2023. 05. 09. 9:48:38 VERBOSE [JwtAuthGuard] Authorized. Guard was disabled for path.
[Nest] 17352  - 2023. 05. 09. 9:48:38     LOG [HTTP] 200 GET /api/nodes/db678827-69ad-4eff-87a5-ae08575d9040/script
[Nest] 17352  - 2023. 05. 09. 9:48:40   DEBUG [AgentService] Agent key: -----BEGIN PGP PUBLIC KEY BLOCK-----
....
-----END PGP PUBLIC KEY BLOCK-----
[Nest] 17352  - 2023. 05. 09. 9:48:40     LOG [AgentService] Agent joined with id: db678827-69ad-4eff-87a5-ae08575d9040, key: true
[Nest] 17352  - 2023. 05. 09. 9:48:40   DEBUG [AgentService] Agents: 1
[Nest] 17352  - 2023. 05. 09. 9:48:40   DEBUG [AgentService] Agent id: db678827-69ad-4eff-87a5-ae08575d9040, open: true
[Nest] 17352  - 2023. 05. 09. 9:48:40   DEBUG [AgentService] Deployments: 0
[Nest] 17352  - 2023. 05. 09. 9:48:40   DEBUG [AgentService] Watchers: 0
[Nest] 17352  - 2023. 05. 09. 9:48:40   DEBUG [AgentService] Log streams: 0
```

😭😭😭

**Review by robot9706 — APPROVED** [review]
🔫

**Pull request merged**